baremetal/metal3machine_manager.go

```diff
@@ -808,13 +808,18 @@ func (m *MachineManager) chooseHost(ctx context.Context) (*bmh.BareMetalHost, *p
 		default:
 			continue
 		}
-		// continue if BaremetalHost is paused
+
+		// continue if BaremetalHost is paused or BaremetalHost is unhealthy
```

Comment:
Maybe the comment also should be updated? since you are checking `[capm3.UnhealthyAnnotation]`

Reply:
but otherwise lgtm